actgen/agents/directed_qnet.py

```diff
@@ -101,7 +102,7 @@ def get_current_q_vals(s):
                     new_q_values = self.forward(qnet_input.to(self.params['device'])).squeeze()  # (n_actions)
                     assert new_q_values.shape == (len(actions), )
                 # update q_delta matrix
-                q_delta[s_idx, a_idx, :] = new_q_values.detach().numpy().squeeze() - original_q_values.detach().numpy().squeeze()
+                q_delta[s_idx, a_idx, :] = new_q_values.detach().to(torch.device('cpu')).numpy().squeeze() - original_q_values.detach().to(torch.device('cpu')).numpy().squeeze()
```

Comment:
I'd prefer `.cpu()` here over `.to(torch.device('cpu'))`. I'm also not sure if it's necessary if you're calling `numpy()` right afterwards

Reply:
When I tried to run it, I think torch doesn't let me convert to `.numpy()` when the tensor is on the GPU.